cmd/kubelet/app/server.go

```diff
@@ -334,6 +334,12 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.Dependencies) (err error) {
 		var clientCertificateManager certificate.Manager
 		if err == nil {
 			if s.RotateCertificates && utilfeature.DefaultFeatureGate.Enabled(features.RotateKubeletClientCertificate) {
+				if clientConfig.CertFile == "" {
```

Comment:
This doesn’t seem like the right place to apply defaults. I’d expect paths to be resolved prior to run() being called

Reply:
Setting these to hardcoded paths is not correct. The data and paths come from the kubeconfig file fed to the kubelet

Reply:
I agree, it is quick and not so nice hack, but it follows same pattern as InitializeTLS() if certificate rotation is disabled.

Reply:
the default paths in InitializeTLS have caused more problems than they have helped. we won't move to that pattern. the core issue seems to be in how `--kubeconfig` files with embedded cert data (rather than a cert file reference) interact with bootstrapping

Reply:
I agree, that is not the clean solution, just quick workarround that was supposed to unblock upgrades 1.7.x->1.8.0. The idea was that for 1.8.1 we should be able to find better solution.